up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -25,332 +24,97 @@ option java_multiple_files = true;
-message UpdatePropertyRequest {
-  // the uri for the node whos property will be updated
-  string uri = 1;
-
-  // The name of the property that is to be updated
-  string property = 2;
-
-  // The value to set in the property
-  PropertyValue value = 3;
-}
-
-// The (empty) response message representing the successful execution of the UpdateProperty operation.
-message UpdatePropertyResponse {}
-
-// Node Change Notification Message.
-// When uEs call RegisterForNotifications(), a Notification message is sent when the node either
-// changes, added, or removed.
-message Notification {
-  // The URI to the node that has changed
-  string uri = 1;
-
-  // the URI of the parent node (if it was affected)
-  optional string parent_uri = 2;
-
-  // The operation performed on said Node
-  Operation operation = 3;
-
-  // Operation
-  enum Operation {
-    INVALID = 0;    // Invalid
-    UPDATE = 1;     // Updated
-    ADD = 2;        // Added to the parent
-    REMOVE = 3;     // Removed
-  }
-
-  // Time-to-live: How long (ms) the information should be live for in the database.
-  // A value of -1 means lives forever.
-  optional  int32 ttl = 4;
-
-  // uDiscovery resource that it serves (per SDV-202 definition): database
-  enum Resources { nodes = 0; }
-}
-
-
-// Observer Identification Information
-message ObserverInfo {
-  // Fully qualified URI for the Observer who is registering to receive the
-  // notifications from uDiscovery ex. `//vcu.VIN/com.gm.app.hartley`
-  string uri = 1;
-}
-
-
-// Passed to the RegisterForNotifications() and UnregisterForNotifications()
-// APIs this message includes the list of one or more node addresses we would like
-// to receive updates for as well as information about the caller so the notification
-// can be routed to the right destination.
-message NotificationsRequest {
-  // A list of one or more Node URIs to receive notifications for
-  repeated string uris = 1;
-
-  // Observer's identification information
-  ObserverInfo observer = 2;
+// Message that stores the metadata about a topic and the topic URI so 
+// it can be passed back in GetServiceTopicsResponse.
+message ServiceTopicInfo {
+  // The URI of the topic
+  uprotocol.v1.UUri topic = 1;
+  
+  // The topic metadata
+  UServiceTopic info = 2;
 
-  // How deep in the node tree should the notifications be sent for. A value of -1 or if
-  // the field is not present in the message, signifies that changes to any child nodes will trigger
-  // a Notification event. A value of 0 returns only the parent node. Any other value specified is
-  // the depth of child nodes to receive notifications for.
-  optional int32 depth = 3;
+  // How long the topic metadata is valid for in seconds.
+  // If the metadata is older than this value, the client MAY re-fetch the metadata.
+  // If the field is missed or the value is 0, the client MAY assume the metadata lives forever.
+  optional uint32 ttl = 3;
```

Comment:
IMHO we do not need to make it optional then. The default value of an int field is `0` by (proto3) definition.